### DISCUSSION TITLE
"url(#paint0_linear)"  bug

### DISCUSSION THREAD

**Discussion by claudia-meneghesso** [tailwindlabs/tailwindcss#4583] [Help]
Hi there,

I just noticed that SVGs exported from Figma which have a linear gradient with a '0' character in their id are not displayed. 

Ex. "url(#paint0_linear)"  will not work, unless you change the URL ref and the id to "#paint1_linear".

A bit odd. Any reason why?

**Comment by VicGUTT** [comment]
Hey again ^^

Seems odd indeed. Don't think I have ever encountered this myself. I know leading numbers in IDs ("#1abc") may sometimes cause problems.

Would you be able to share the SVG in something like [play.tailwindcss.com](play.tailwindcss.com) ?

**Reply by claudia-meneghesso** [reply]
Hy VicGUTT,

thanks a lot for your prompt reply!

Here we go: 

https://play.tailwindcss.com/Aqc2fAqtyd

Although, it seems not to work at all in this playground.

**Reply by VicGUTT** [reply]
Hey, my pleasure.

I have given the provided demo a try, and after a few tweaks it seems perfectly fine. Here's my end result : [https://play.tailwindcss.com/9pFKA5bTKK](https://play.tailwindcss.com/9pFKA5bTKK).

Although I have no idea what the SVG is supposed to look like, so maybe I'm crying victory too soon.

Anyway I really didn't do much. What I did was mostly convert JSX back to HTML in order for it to work on the playground. As you noticed yourself, the SVG wasn't rendering properly. That is due to the fact that you seem to be using JSX as your templating system and had to transform valid SVG attributes like `stop-color` to `stopColor`, but when you pasted the SVG code to the playground you forgot to transform those attributes back.

Has for the issue at hand, the "0" in `url(#paint0_linear)` does not seem to cause any issues, unless I'm missing something.

Hope I help clarify things a little :)

**Reply by claudia-meneghesso** [reply]
Hi Vic, 

Your playground example works as expected. However, I am afraid this might not be the reason. I only provided a small example in my playground, but, in the project I am working there are actually three shapes and 3 gradients, with only the url(#paint0_linear) id not working properly. A colleague suggested that, if another gradient is present on another file but with the same id, they may clash. So let's say, I have 2 components, each with an SVG in it but totally distinct. If they have the same gradient it won't work. I still have to properly test this to be 100% sure, but I will and I will get back at you. Currently working with Tailwind on Gatsby.
